### PR TITLE
Pre-import some built-in modules in PyPy

### LINKED ISSUE #1201: Some builtin modules should be preimported in site.py on PyPy
For PyPy, some built-in modules should be pre-imported in `site.py` to match CPython, or some programs may not be working properly, for example, the Python Console in PyCharm.

What PyPy does currently(L532-539, L545): 

https://bitbucket.org/pypy/pypy/src/314ddd2d83c5e467aed4f64ab4b8b2fc0047540b/lib-python/2.7/site.py?at=default&fileviewer=file-view-default#site.py-532:539,545

The original commits:

https://bitbucket.org/pypy/pypy/commits/1c6dc3e6e70c6fadad60ba9c3990b49c8aaf16bd
https://bitbucket.org/pypy/pypy/commits/f160281ab6613617a49a73129f81d7ef3540755a

### PR DESCRIPTION
**Pull request by microdog**
Pre-import some built-in modules while running in PyPy to match what CPython does.
This fixes #1201 .

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [ ] updated/extended the documentation
- [x] added news fragment in ``docs/changelog`` folder

### CODE REVIEW

File: tests/test_virtualenv.py [hunk 1]
```diff
@@ -471,3 +474,40 @@ def test_create_environment_with_local_https_pypi(tmpdir):
...
+
+def check_pypy_pre_import():
+    import sys
+
+    missing_str = "missing %r in sys.modules"
+
+    assert "encodings" in sys.modules, missing_str % ("encodings",)
+
+    if "exceptions" in sys.builtin_module_names:
+        assert "exceptions" in sys.modules, missing_str % ("exceptions",)
```
Comment (gaborbernat) [inline tests/test_virtualenv.py]: we prefer the usage of the format method over the ``%`` operator

File: virtualenv_embedded/site.py [hunk 2]
```diff
@@ -738,7 +738,9 @@ def write_history():
     def import_builtin_stuff():
         """PyPy specific: pre-import some built-in modules to match CPython."""
         import encodings
-        import exceptions
+
+        if "exceptions" in sys.builtin_module_names:
```
Comment (gaborbernat) [inline virtualenv_embedded/site.py]: would be nice to refer to some kind of pypy documentation/link about why this is needed

File: tests/test_virtualenv.py [hunk 3]
```diff
@@ -471,3 +474,40 @@ def test_create_environment_with_local_https_pypi(tmpdir):
...
+
+    if "zipimport" in sys.builtin_module_names:
```
Comment (gaborbernat) [inline tests/test_virtualenv.py]: these two probably should be in a for loop instead

File: tests/test_virtualenv.py [hunk 4]
```diff
@@ -471,3 +474,40 @@ def test_create_environment_with_local_https_pypi(tmpdir):
...
+        assert "zipimport" in sys.modules, missing_str % ("zipimport",)
+
+
+@pytest.mark.skipif("platform.python_implementation() != 'PyPy'")
+def test_pypy_pre_import(tmpdir):
```
Comment (gaborbernat) [inline tests/test_virtualenv.py]: prefer tmp_path over tmpdir

**Comment by microdog** [comment]
Changelog added. 😄

**Comment by gaborbernat** [comment]
thanks

**Pull request merged**